SocialCareCaseViewerApi.Tests/V1/IntegrationTests/UpdateWorkerDetails.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Bogus;
+using FluentAssertions;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using SocialCareCaseViewerApi.V1.Boundary.Requests;
+using SocialCareCaseViewerApi.V1.Boundary.Response;
+using SocialCareCaseViewerApi.V1.Domain;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+
+namespace SocialCareCaseViewerApi.Tests.V1.IntegrationTests
+{
+    [TestFixture]
+    public class UpdateWorkerDetails : IntegrationTestSetup<Startup>
+    {
+        [Test]
+        public async Task UpdateWorkerWithNewTeamReturnsTheOnlyTheUpdatedTeam()
+        {
+            // Get request to get an existing worker
+            var existingWorker = IntegrationTestHelpers.SetupExistingWorker(DatabaseContext);
+
+            var newTeam = IntegrationTestHelpers.CreateAnotherTeam(DatabaseContext, existingWorker.ContextFlag);
+
+            var getWorkersUri = new Uri($"/api/v1/workers?email={existingWorker.Email}", UriKind.Relative);
+
+            var getWorkersResponse = await Client.GetAsync(getWorkersUri).ConfigureAwait(true);
+            getWorkersResponse.StatusCode.Should().Be(200);
+
+            var initialContent = await getWorkersResponse.Content.ReadAsStringAsync().ConfigureAwait(true);
+            var initialWorkerResponse = JsonConvert.DeserializeObject<List<WorkerResponse>>(initialContent).ToList();
+
+            initialWorkerResponse.Count.Should().Be(1);
+            initialWorkerResponse.Single().Teams.Single().Id.Should().NotBe(newTeam.Id);
+            initialWorkerResponse.Single().Teams.Single().Name.Should().NotBe(newTeam.Name);
+
+            // Patch request to update team
+
+            var newTeamRequest = new WorkerTeamRequest { Id = newTeam.Id, Name = newTeam.Name };
+
+            var patchUri = new Uri("/api/v1/workers", UriKind.Relative);
+            var patchRequest = IntegrationTestHelpers.CreatePatchRequest(existingWorker, newTeamRequest);
+
+            var serializedRequest = JsonSerializer.Serialize(patchRequest);
+            var requestContent = new StringContent(serializedRequest, Encoding.UTF8, "application/json");
+
+            var patchWorkerResponse = await Client.PatchAsync(patchUri, requestContent).ConfigureAwait(true);
+            patchWorkerResponse.StatusCode.Should().Be(204);
+
+            // Get request to check team has been updated
+
+            var getUpdatedWorkersResponse = await Client.GetAsync(getWorkersUri).ConfigureAwait(true);
+            getUpdatedWorkersResponse.StatusCode.Should().Be(200);
+
+            var updatedContent = await getUpdatedWorkersResponse.Content.ReadAsStringAsync().ConfigureAwait(true);
+            var updatedWorkerResponse = JsonConvert.DeserializeObject<List<WorkerResponse>>(updatedContent).ToList();
+
+            updatedWorkerResponse.Count.Should().Be(1);
+            updatedWorkerResponse.Single().Teams.Count.Should().Be(1);
```

Comment:
I was expecting this line to fail and error with a message related to 2 teams being found.
When you look at the response when we update a worker and view their details in the UI, the response from the API has a more than one element in the teams list for the worker.

Not sure why I can't recreate the same process here 🤔 . Might have missed something?

Reply:
Have been able to recreate this in 03ff05b